pyfar/dsp/dsp.py

```diff
@@ -227,6 +227,202 @@ def spectrogram(signal, dB=True, log_prefix=20, log_reference=1,
+
+    This function calls `scipy.signal.windows.get_window` to create the
+    window.
+    Window types:
+    - ``boxcar``
+    - ``triang``
+    - ``blackman``
+    - ``hamming``
+    - ``hann``
+    - ``bartlett``
+    - ``flattop``
+    - ``parzen``
+    - ``bohman``
+    - ``blackmanharris``
+    - ``nuttall``
+    - ``barthann``
+    - ``kaiser`` (needs beta, see :py:func:`~pyfar.dsp.kaiser_window_beta`)
+    - ``gaussian`` (needs standard deviation)
+    - ``general_gaussian`` (needs power, width)
+    - ``dpss`` (needs normalized half-bandwidth)
+    - ``chebwin`` (needs attenuation)
+    - ``exponential`` (needs center, decay scale)
+    - ``tukey`` (needs taper fraction)
+    - ``taylor`` (needs number of constant sidelobes,
+      sidelobe level)
+    If the window requires no parameters, then `window` can be a string.
+    If the window requires parameters, then `window` must be a tuple
+    with the first argument the string name of the window, and the next
+    arguments the needed parameters.
+    """
+    # Check input
+    if not isinstance(signal, pyfar.Signal):
+        raise TypeError("The parameter signal has to be of type: Signal.")
+    if shape not in ('symmetric', 'left', 'right'):
+        raise ValueError(
+            "The parameter shape has to be 'symmetric', 'left' or 'right'.")
+    if not isinstance(truncate, bool):
+        raise TypeError("The parameter truncate has to be of type: bool.")
+    if not isinstance(length, (list, type(None))):
+        raise TypeError(
+            "The parameter length has to be of type list or None.")
+
+    # Check length
+    if length is not None:
+        if length != sorted(length):
+            raise ValueError("Values in length need to be in ascending order.")
+        # Convert to samples
+        if unit == 's':
+            length = [round(li*signal.sampling_rate) for li in length]
+        elif unit == 'ms':
+            length = [round(li*signal.sampling_rate/1e3) for li in length]
+        elif unit == 'samples':
+            length = [int(li) for li in length]
+        else:
+            raise ValueError(f"unit is {unit} but has to be"
+                             f" 'samples', 's' or 'ms'.")
+        # Check window size
+        if length[-1] > signal.n_samples:
+            raise ValueError(
+                "Values in length require window to be longer than signal.")
+
+    # Create window
+    # win_start and win_stop define the first and last sample of the window
+    if length is None:
+        win_samples = signal.n_samples
+        win = sgn.windows.get_window(window, win_samples, fftbins=False)
+        win_start = 0
+        win_stop = signal.n_samples-1
+    elif len(length) == 2:
+        if shape == 'symmetric':
+            win_samples = length[1]-length[0]+1
+            win = sgn.windows.get_window(window, win_samples, fftbins=False)
+            win_start = length[0]
+            win_stop = length[1]
```

Comment:
Could we also have a window that is symmetric with respect to t=0? This is usually very useful for non minimum-phase IRs without a linear phase component.

Reply:
Good suggestion! Suggestion: single value of length (len(length)==1 and shape=='symmetric')

Reply:
Good Point! Added option shape='symmetric_zero', also as default

Reply:
👍